server/src/main/java/org/opensearch/action/admin/cluster/node/stats/NodeStats.java

```diff
@@ -126,6 +127,9 @@ public class NodeStats extends BaseNodeResponse implements ToXContentFragment {
     @Nullable
     private ClusterManagerThrottlingStats clusterManagerThrottlingStats;
 
+    @Nullable
+    private WeightedRoutingStats weightedRoutingStats;
```

Comment:
Since this is only related to Search, does it make sense to be placed in `SearchStats` instead ?

Reply:
Yeah makes sense. Tried to go through the flow but looks like since failopen stats is captured inside `FetchPhase`, there isn't an easy way to capture this since the  current implementation is just looking for fetchphase https://github.com/opensearch-project/OpenSearch/blob/main/server/src/main/java/org/opensearch/index/shard/SearchOperationListener.java. Will require some deeper understanding in this as we need to capture fail open stats in all the actions modified in https://github.com/opensearch-project/OpenSearch/blob/main/server/src/main/java/org/opensearch/index/shard/SearchOperationListener.java. Do you have any pointers on this?